### ISSUE TITLE
Transaction preview leaks signed transaction to 3rd website http://coinb.in

### ISSUE THREAD

**Issue by Transisto** [BlueWallet/BlueWallet#5017]
Surprised how it seems so hard to find code that convert a raw transaction into human readable text that we need to have our signed transaction sent to a 3rd party website.

**Comment by Overtorment** [comment]
its leaked only when you actually press "view on coinb.in" button

**Comment by Transisto** [comment]
Why not bundle it and run it locally? It's open source code. 

"You can even [download](https://github.com/OutCast3k/coinbin/archive/master.zip) this page and host it yourself or run it offline!" -Coinbin

**Comment by ncoelho** [comment]
Because this is only useful to verify against a 3rd party.

**Comment by Transisto** [comment]
> Because this is only useful to verify against a 3rd party.

There is no point validating this with a third party.,This isn't a block explorer to validate chain state it's only to be able to know what utxo have been selected as part of the transaction.

**Comment by ncoelho** [comment]
Not at all the point of this. Or the threat model in place here.

This is to validate the transaction signed against a different source so
you don’t have to trust bluewallet.

If you don’t like this feature don’t use it.

**Comment by Transisto** [comment]
The source of the problem is that BW doesn't have the feature to display important transaction details before signing or before broadcasting.

Sending a signed transaction to an online third party to verify that BW tx signing code, arguably the most important part of the wallet is not buggy is ridiculous.

**Comment by ncoelho** [comment]
Agree, feel free to open a PR. Closing the issue as there is no leak from the app.

**Comment by Transisto** [comment]
If you want to keep the online link as a stopgap from implementing a proper preview feature in the wallet then the wallet should protect the users privacy by having them confirm that they their signed transaction will be published to a third-party which may;

Broadcast it to the network,
Keep a copy of it along with the IP of the user.
